packages/preferences/src/browser/views/components/preference-select-input.ts

```diff
@@ -17,29 +17,60 @@
 import { PreferenceLeafNodeRenderer, PreferenceNodeRenderer } from './preference-node-renderer';
 import { injectable, interfaces } from '@theia/core/shared/inversify';
 import { JSONValue } from '@theia/core/shared/@phosphor/coreutils';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { SelectComponent, SelectOption } from '@theia/core/lib/browser/widgets/select-component';
 import { Preference } from '../../util/preference-types';
 import { PreferenceLeafNodeRendererContribution } from './preference-node-renderer-creator';
+import * as React from '@theia/core/shared/react';
+import * as ReactDOM from '@theia/core/shared/react-dom';
 
 @injectable()
-export class PreferenceSelectInputRenderer extends PreferenceLeafNodeRenderer<JSONValue, HTMLSelectElement> {
+export class PreferenceSelectInputRenderer extends PreferenceLeafNodeRenderer<JSONValue, HTMLDivElement> {
+
+    protected readonly onDidChangeEmitter = new Emitter<number>();
+
+    protected get onDidSelectedChange(): Event<number> {
+        return this.onDidChangeEmitter.event;
+    }
 
     protected get enumValues(): JSONValue[] {
         return this.preferenceNode.preference.data.enum!;
     }
 
+    protected get selectComponentOptions(): SelectOption[] {
+        const items: SelectOption[] = [];
+        const values = this.enumValues;
+        const defaultValue = this.preferenceNode.preference.data.default;
+        for (let i = 0; i < values.length; i++) {
+            const value = `${values[i]}`;
+            const detail = defaultValue === value ? 'default' : undefined;
```

Comment:
Since `enum` values can be objects, I think this should be a deep equality check (`PreferenceProvider.deepEqual`).